app/server/lib/DocSnapshots.ts

```diff
@@ -6,80 +6,83 @@ import {ExternalStorage} from 'app/server/lib/ExternalStorage';
 import log from 'app/server/lib/log';
 import * as fse from 'fs-extra';
 import * as moment from 'moment-timezone';
+import {integerParam} from 'app/server/lib/requestUtils';
 
 /**
  * A subset of the ExternalStorage interface, focusing on maintaining a list of versions.
  */
 export interface IInventory {
-  getSnapshotWindow?: (key: string) => Promise<SnapshotWindow|undefined>;
-  versions(key: string): Promise<ObjSnapshotWithMetadata[]>;
-  remove(key: string, snapshotIds: string[]): Promise<void>;
+    getSnapshotWindow?: (key: string) => Promise<SnapshotWindow | undefined>;
+
+    versions(key: string): Promise<ObjSnapshotWithMetadata[]>;
+
+    remove(key: string, snapshotIds: string[]): Promise<void>;
 }
 
 /**
  * A utility for pruning snapshots, so the number of snapshots doesn't get out of hand.
  */
 export class DocSnapshotPruner {
-  private _closing: boolean = false;                        // when set, should ignore prune requests
-  private _prunes: KeyedOps;
-
-  // Specify store to be pruned, and delay before pruning.
-  constructor(private _ext: IInventory, _options: {
-    delayBeforeOperationMs?: number,
-    minDelayBetweenOperationsMs?: number
-  } = {}) {
-    this._prunes = new KeyedOps((key) => this.prune(key), {
-      ..._options,
-      retry: false,
-      logError: (key, failureCount, err) => log.error(`Pruning document ${key} gave error ${err}`)
-    });
-  }
-
-  // Shut down.  Prunes scheduled for the future are run immediately.
-  // Can be called repeated safely.
-  public async close() {
-    this._closing = true;
-    this._prunes.expediteOperations();
-    await this.wait();
-  }
-
-  // Wait for all in-progress prunes to finish up in an orderly fashion.
-  public async wait() {
-    await this._prunes.wait(() => 'waiting for pruning to finish');
-  }
-
-  // Note that a document has changed, and should be pruned (or repruned).  Pruning operation
-  // done as a background operation.  Returns true if a pruning operation has been scheduled.
-  public requestPrune(key: string): boolean {
-    // If closing down, do not accept any prune requests.
-    if (!this._closing) {
-      // Mark the key as needing work.
-      this._prunes.addOperation(key);
+    private _closing: boolean = false;                        // when set, should ignore prune requests
```

Comment:
Sorry about this @softwareguru90 but is there any way you could avoid these whitespace changes? It makes it hard to check for actual changes in this code. You may need to change a setting in your IDE. You'll know you've got things right if `git diff main` in a table shows your changes and only your changes.